pilot/pkg/config/kube/crd/controller.go

```diff
@@ -58,25 +62,41 @@ var (
 		Help: "Errors converting k8s CRDs",
 	}, []string{"name"})
 
+	k8sResourceVersions = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "pilot_k8s_resource_versions",
+		Help: "Applied CRD resource versions",
+	}, []string{"gvk", "name", "namespace", "resourceVersion", "instanceID"})
```

Comment:
Is namespace needed as you said instanceID is of format pod namespace:pod name?

Reply:
`name` and `namespace` identify the resource (e.g. details-default VirtualService). `instanceID` identifies the workload that applied the resource (e.g. Pilot)